src/foam/nanos/controller/ApplicationController.js

```diff
@@ -1042,6 +1044,16 @@ foam.CLASS({
           // TODO: add support for being able to pick if multiple menus have the same obj
           // menus.push(menuDAOs[i]);
       }
+    },
+    function logAnalyticEvent(evtName, evtTraceId, evtSessionId, evtExtra) {
+      this.__subContext__.analyticEventDAO.put(foam.nanos.analytics.AnalyticEvent.create(
+        {
+          name: evtName,
+          sessionId: evtSessionId,
+          traceId: evtTraceId,
+          extra: evtExtra
+        }
+      ));
```

Comment:
```suggestion
      ), this);
```

Reply:
Actually, the applicationController should import 'analyticEventDAO' and require the AnalyticEvent class, this should just be 
`this.analyticEventDAO?.put(this.AnalyticEvent.create({....}))`

Reply:
updated